python/morphology.cpp

```diff
@@ -142,7 +142,11 @@ void register_morphology(pybind11::module& m) {
                 throw pyarb_error(util::pprintf("can't open file '{}'", fname));
             }
             try {
-                return arborio::load_swc_arbor(arborio::parse_swc(fid));
+                auto data = arborio::parse_swc(fid);
+                if (fid.rdbuf()->in_avail()) {
```

Comment:
Should trailing white space/blank lines be permitted?
If so, could have e.g.
```
auto data = arborio::parse_swc(fid);
if (!(fid >> std::ws).eof()) { throw …; }
```

Either way, the test and throw across the different load functions could be encapsulated into a single function to avoid duplication of the error message and logic.

Reply:
Okay here's what I think we can do: 
1. I'll encapsulate the checking and error message as you suggested.
2. `parse_swc` checks if the line is empty before parsing a record, using `istream::peek()`; if empty, it stops further parsing and leave the stream in a 'good' state. With this, `swc1 = parse_swc(fs); fs >> std::ws; swc2 = parse_swc(fs);` should work providing there are no other errors. 
3. If we read a non-blank line that is not parsed successfully as an SWC record, the fail bit is set on the stream: an error is raised in the python wrapper, we can't continue reading from the stream